static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -4,127 +4,142 @@ import capitalize from 'lodash/capitalize';
 import pick from 'lodash/pick';
 
 import {t, tct} from 'sentry/locale';
-import {ExternalActorMapping, Integration, Organization} from 'sentry/types';
+import {ExternalActorMapping, Integration} from 'sentry/types';
+import {getExternalActorEndpointDetails} from 'sentry/utils/integrationUtil';
 import {FieldFromConfig} from 'sentry/views/settings/components/forms';
 import Form from 'sentry/views/settings/components/forms/form';
+import FormModel from 'sentry/views/settings/components/forms/model';
 import {Field} from 'sentry/views/settings/components/forms/type';
 
-type Props = Pick<Form['props'], 'onSubmitSuccess' | 'onCancel'> &
-  Partial<Pick<Form['props'], 'onSubmit'>> & {
-    organization: Organization;
-    integration: Integration;
-    mapping?: ExternalActorMapping;
-    type: 'user' | 'team';
-    baseEndpoint?: string;
-    sentryNamesMapper: (v: any) => {id: string; name: string}[];
-    url: string;
-    onResults?: (data: any) => void;
-  };
+type Props = Pick<Form['props'], 'onCancel' | 'onSubmitSuccess' | 'onSubmitError'> & {
+  integration: Integration;
+  mapping?: ExternalActorMapping;
+  type: 'user' | 'team';
+  getBaseFormEndpoint: (mapping?: ExternalActorMapping) => string;
+  sentryNamesMapper: (v: any) => {id: string; name: string}[];
+  dataEndpoint: string;
+  onResults?: (data: any, mappingKey?: string) => void;
+  isInline?: boolean;
+  mappingKey?: string;
+};
 
 export default class IntegrationExternalMappingForm extends Component<Props> {
+  model = new FormModel();
+
   get initialData() {
     const {integration, mapping} = this.props;
-
     return {
-      externalName: '',
-      userId: '',
-      teamId: '',
-      sentryName: '',
       provider: integration.provider.key,
       integrationId: integration.id,
-      ...pick(mapping, ['externalName', 'userId', 'sentryName', 'teamId']),
+      ...pick(mapping, ['id', 'externalName', 'sentryName', 'userId', 'teamId']),
```

Comment:
The `pick()` function "unions" together the object and the list of keys. Since this list is every key you can skip the `pick()`.
```tsx
return {
      provider: integration.provider.key,
      integrationId: integration.id,
      ...mapping,
    };
```

Reply:
lol I didn't even notice as I kept adding stuff to this. Great catch :+1: